tidy3d/components/heat_charge/doping.py

```diff
@@ -0,0 +1,98 @@
+"""File containing classes required for the setup of a DEVSIM case."""
+
+from abc import ABC
+
+import numpy as np
+import pydantic.v1 as pd
+
+from ..base import Tidy3dBaseModel, cached_property
+from ..types import Bound, Union
+
+
+class AbstractDopingBox(ABC, Tidy3dBaseModel):
```

Comment:
One possibility you can consider is to inherit actual class `Box` here. It follows how we typically define rectangular regions in Tidy3d (center + size), but still has convenience method `from_bounds` https://github.com/flexcompute/tidy3d/blob/marc/dd-devsim-doping/tidy3d/components/geometry/base.py#L1889